examples/platforms/gp712/uart-socket.c

```diff
@@ -0,0 +1,348 @@
+
+                PlatSocketRx(readLen, buffer, clientSocket->socketId);
+
+                res = write(PlatSocketPipeFd [1], &someByte, 1); //[1] = write fd
+                (void)(res);
+            }
+        }
+    }
+    clientSocket->isValid = 0;
+    qorvoPlatUnRegisterPollFunction(clientSocket->socketId);
+    close(clientSocket->socketId);
+
+    PlatSocketClose(clientSocket->socketId);
+
+    return NULL;
+}
+
+void PlatSocketRxNewConn(uint8_t id)
+{
+    //Find first non-valid client in list - add here
+        if (PlatSocketConnection.isValid == 0)
+        {
+            //Add new client to client list
+            socklen_t len;
+            len = sizeof(PlatSocketConnection.addr);
+            int retval = accept (id, (struct sockaddr *)&PlatSocketConnection.addr, (socklen_t*) &len);
+
+            if (retval >= 0)
+            {
+                int retErr;
+                PlatSocketConnection.socketId = retval;
+                retErr = pthread_create( &PlatSocketConnection.rfReadThread,
+                                        NULL,
+                                        PlatSocketReadThread,
+                                        &PlatSocketConnection);
+                if (retErr)
+                {
+                    close(PlatSocketConnection.socketId);
+                }
+                else
+                {
+                    PlatSocketConnection.isValid = 1;
+                }
+            }
+        }
+        else
+        {
+
+            int tempfd;
+            tempfd = accept (id, (struct sockaddr *)NULL, NULL);
+            if (tempfd >= 0)
+            {
+                close(tempfd);
+            }
+        }
+}
+
+/*****************************************************************************
+ *                    Public Function Definitions
+ *****************************************************************************/
+
+void PlatSocketInit(void)
+{
+    int res;
+    memset(&PlatSocketConnection, 0, sizeof(PlatSocketConnection));
+
+    // in case we are a server, setup listening for client
+    PlatServerSocketId = PlatSocketListenForClients();
+    qorvoPlatRegisterPollFunction(PlatServerSocketId, PlatSocketRxNewConn);
+
+    // hack
+    res = pipe(PlatSocketPipeFd);
+    (void)(res);
+    qorvoPlatRegisterPollFunction(PlatSocketPipeFd [0], PlatSocketRxSignaled);
+}
+
+void platformUartRestore(void)
+{
+    PlatSocketDeInit();
+}
+
+void PlatSocketDeInit(void)
+{
+    qorvoPlatUnRegisterPollFunction(PlatServerSocketId);
+    close(PlatServerSocketId);
+    qorvoPlatUnRegisterPollFunction(PlatSocketPipeFd[0]);
+    close(PlatSocketPipeFd[0]);
+    close(PlatSocketPipeFd[1]);
+    close(PlatSocketConnection.socketId);
+}
+
+int PlatSocketTxData(uint16_t length, uint8_t *pData, uint32_t socketId)
+{
+    int result = -1;
+
+    //All sockets
+    if(PlatSocketConnection.isValid)
+    {
+        if((int)PlatSocketConnection.socketId == (int)socketId)
```

Comment:
Both sides of this conditional are originally `uint32_t`, so why are they both being cast to `int` before comparison?